### ISSUE TITLE
Cronjob failing due to not continuously assigned IDs in jobs table

### ISSUE THREAD

**Issue by MarioSchymura** [owncloud/core#18295]
Hi there.

After upgrading to OC 8.1 and seperatly updating my installed apps, I did recognize that my cron.php was not handling all exiting jobs anylonger. 

The main reason for this was a gap within the continuously assigned IDs in jobs table. Probably any app upgrade was the reason for this, but this is not really replicable for me. This gap within the given IDs, that is ID 5 was present, but the next ID was 9, lead to a stop of job processing at ID 5. 

It seems that the current implementation of "getLastJob()" in "joblist.php" return the last processed job's ID, but the method "getNext()" is currently not able to skip the gap of IDs in the table, so it stops without processing the rest of existing jobs. 

I solved it by re-assigning consistent IDs in MySql, but I think the method "getNext()" shall be able to handle such a gap of IDs, because it is obviously possible to create such situation using the WebClient.

**Comment by DeepDiver1975** [comment]
@icewind1991

**Comment by icewind1991** [comment]
I can't reproduce this issue, I setup a test case which creates and deletes some jobs to create a situation where there is a gap in job ids but the next job is returned as expected.

Also looking at the code to select the next job I cant see any reason why it shouldn't handle a gap correctly as long as the job id's are ordered correctly

**Comment by MarioSchymura** [comment]
Hi there

I apologize for not describing the failure situation correctly. I did ran into the error again and i now know the real cause. I did deactive an app and it's jobs remained in the job table.

Then, when "getNext()" in "joblist.php" gets called, the method exits because of the call to "buildJob" exits with null. Obviously the check 

"if (!class_exists($class))" 

evaulates to true, so the return value is null. That leads to an Id being null and therefore the processing of jobs did fail. 

I'm not quite sure, but this may only be the case when using webCron, which I'm sadly forced to use due to my providers limitations.
